# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/lib/wasm.js
# Commit: 72bf9c871ded
# Full Hash: 72bf9c871ded006f56f09b24d0723365516445bb
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 5
# Description:
#   Bug 1692065 - wasm: Add tagging for 31-bit integers and i31ref. r=yury
#   
#   This commit adds a tag 0x1 for 31-bit integers in AnyRef. Any JS value
#   that is losslessly representable (without coercion) in this range is
#   stored without boxing. This is chosen to line up exactly with how the
# ==============================================================================

diff -r 28feda14fdec -r 72bf9c871ded js/src/jit-test/lib/wasm.js
--- a/js/src/jit-test/lib/wasm.js	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/jit-test/lib/wasm.js	Fri Aug 04 14:06:50 2023 +0000
@@ -443,19 +443,55 @@
     WasmArrayrefValues.push(newArray());
 }
 
+let WasmGcObjectValues = WasmStructrefValues.concat(WasmArrayrefValues);
+
 // Valid values for eqref
 let WasmEqrefValues = [...WasmStructrefValues, ...WasmArrayrefValues];
 
+// Valid values for i31ref
+let MinI31refValue = -1 * Math.pow(2, 30);
+let MaxI31refValue = Math.pow(2, 30) - 1;
+let WasmI31refValues = [
+  // first four 31-bit signed numbers
+  MinI31refValue,
+  MinI31refValue + 1,
+  MinI31refValue + 2,
+  MinI31refValue + 3,
+  // five numbers around zero
+  -2,
+  -1,
+  0,
+  1,
+  2,
+  // last four 31-bit signed numbers
+  MaxI31refValue - 3,
+  MaxI31refValue - 2,
+  MaxI31refValue - 1,
+  MaxI31refValue,
+];
+
 // Valid and invalid values for anyref
-let WasmAnyrefValues = [...WasmEqrefValues];
+let WasmAnyrefValues = [...WasmEqrefValues, ...WasmI31refValues];
 let WasmNonAnyrefValues = [
     undefined,
     true,
     false,
     {x:1337},
     ["abracadabra"],
-    1337,
     13.37,
+    -0,
+    0x7fffffff + 0.1,
+    -0x7fffffff - 0.1,
+    0x80000000 + 0.1,
+    -0x80000000 - 0.1,
+    0xffffffff + 0.1,
+    -0xffffffff - 0.1,
+    Number.EPSILON,
+    Number.MAX_SAFE_INTEGER,
+    Number.MIN_SAFE_INTEGER,
+    Number.MIN_VALUE,
+    Number.MAX_VALUE,
+    Number.NaN,
     "hi",
     37n,
     new Number(42),